### PR TITLE
UI more changes to save/load draft amendments.

### PR DESCRIPTION
**Pull request by doug-lovett**
Signed-off-by: Doug Lovett <doug@diamante.ca>

*Issue #:* /bcgov/entity#9102

*Description of changes:*
- save draft API call and tests
- load draft API call, update store, and tests
- API setup to save amendment registration
- My registrations table draft amendment start flow, load draft data.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/utils/registration-helper.ts [hunk 1]
```diff
@@ -379,6 +501,92 @@ export async function setupFinancingStatementDraft (stateModel:StateModelIF, doc
   return stateModel
 }
 
+export function setupStateModelFromAmendmentDraft (stateModel:StateModelIF, draft:DraftIF): StateModelIF {
```
Comment (jordiwes) [inline ppr-ui/src/utils/registration-helper.ts:504]: Just to clarify, what is this function for vs. setupAmendmentStatementFromDraft  :)

File: ppr-ui/src/views/amendment/AmendRegistration.vue [hunk 2]
```diff
@@ -170,12 +210,11 @@ export default class AmendRegistration extends Vue {
       this.emitError(financingStatement.error)
     } else {
       // load data into the store
-      const registrationType = RegistrationTypes.find(
-        (reg, index) => {
-          if (reg.registrationTypeAPI === financingStatement.type) {
-            return true
-          }
-        })
+      const registrationType = RegistrationTypes.find((reg, index) => {
```
Comment (jordiwes) [inline ppr-ui/src/views/amendment/AmendRegistration.vue]: Just to clarify, we need the financing statement still, if we are loading a draft amendment?

**Review by jordiwes — APPROVED** [review]
Looks good, I just have a couple of clarifying questions

File: ppr-ui/src/utils/registration-helper.ts (hunk 1 above)
Comment (doug-lovett) [reply ppr-ui/src/utils/registration-helper.ts:504]: setupAmendmentStatementFromDraft makes an API call to get the amendment draft. It then calls setupStateModelFromAmendmentDraft to update the model with the draft data. Splitting into setupStateModelFromAmendmentDraft facilitates unit testing with different scenarios (without making an API call).

File: ppr-ui/src/views/amendment/AmendRegistration.vue (hunk 2 above)
Comment (doug-lovett) [reply ppr-ui/src/views/amendment/AmendRegistration.vue]: Yes, the way the api currently works a draft amendment only stores what is changing. The draft data (when complete) is identical to what is submitted in the registration request.

**Comment by jordiwes** [comment]
You just have some conflicts now :)

**Pull request merged**